src/settings/settings_manager.cpp

```diff
@@ -0,0 +1,227 @@
+    : db_(db), settings_handle_(catalog->GetSettingsHandle()), txn_manager_(txn_manager) {
+  ValidateParams();
+  InitializeCatalog();
+}
+
+void SettingsManager::ValidateParams() {
+  // This will expand to invoke settings_manager::DefineSetting on
+  // all of the settings defined in settings.h.
+  // Example:
+  //   ValidateSetting(Param::port, type::TransientValueFactory::GetInteger(1024),
+  //                  type::TransientValueFactory::GetInteger(65535));
+
+#define __SETTING_VALIDATE__           // NOLINT
+#include "settings/settings_common.h"  // NOLINT
+#include "settings/settings_defs.h"    // NOLINT
+#undef __SETTING_VALIDATE__            // NOLINT
+}
+
+void SettingsManager::ValidateSetting(Param param, const type::TransientValue &min_value,
+                                      const type::TransientValue &max_value) {
+  const ParamInfo &info = db_->param_map_.find(param)->second;
+  if (!ValidateValue(info.value, min_value, max_value)) {
+    SETTINGS_LOG_ERROR(
+        "Value given for \"{}"
+        "\" is not in its min-max bounds",
+        info.name);
+    throw SETTINGS_EXCEPTION("Invalid setting value");
+  }
+}
+
+void SettingsManager::InitializeCatalog() {
+  auto txn = txn_manager_->BeginTransaction();
+  auto column_num = catalog::SettingsHandle::schema_cols_.size();
+
+  for (const auto &pair : db_->param_map_) {
+    const Param param = pair.first;
+    const ParamInfo &info = pair.second;
+
+    catalog::settings_oid_t oid(static_cast<uint32_t>(param));
+    std::vector<type::TransientValue> entry;
+    for (auto i = column_num; i > 0; --i) {
+      // NOLINTNEXTLINE
+      entry.emplace_back(ValueFactory::GetNull(type::TypeId::VARCHAR));
+    }
+
+    entry[static_cast<int>(Index::OID)] = ValueFactory::GetInteger(!oid);
+    entry[static_cast<int>(Index::NAME)] = ValueFactory::GetVarChar(info.name.c_str());
+    entry[static_cast<int>(Index::SHORT_DESC)] = ValueFactory::GetVarChar(info.desc.c_str());
+
+    settings_handle_.InsertRow(txn, entry);
+  }
+
+  txn_manager_->Commit(txn, EmptyCallback, nullptr);
+}
+
+int32_t SettingsManager::GetInt(Param param) {
+  common::SharedLatch::ScopedExclusiveLatch guard(&latch_);
+  return ValuePeeker::PeekInteger(GetValue(param));
+}
+
+double SettingsManager::GetDouble(Param param) {
+  common::SharedLatch::ScopedExclusiveLatch guard(&latch_);
+  return ValuePeeker::PeekDecimal(GetValue(param));
+}
+
+bool SettingsManager::GetBool(Param param) {
+  common::SharedLatch::ScopedExclusiveLatch guard(&latch_);
+  return ValuePeeker::PeekBoolean(GetValue(param));
+}
+
+std::string_view SettingsManager::GetString(Param param) {
+  common::SharedLatch::ScopedExclusiveLatch guard(&latch_);
+  return ValuePeeker::PeekVarChar(GetValue(param));
+}
+
+void SettingsManager::SetInt(Param param, int32_t value, std::shared_ptr<ActionContext> action_context,
+                             setter_callback_fn setter_callback) {
```

Comment:
Why `setter_callback_fn` is snake case whereas all other types and camel case?

Reply:
We were trying to be consistent with transaction_manager.